runtime/jcl/common/java_lang_Class.cpp

```diff
@@ -1888,8 +2061,22 @@ Java_java_lang_Class_getNestMembersImpl(JNIEnv *env, jobject recv)
 			resultObject = POP_OBJECT_IN_SPECIAL_FRAME(currentThread);
 
 			if (NULL == nestMember) {
+				/* If internalFindClassUTF8 fails to find the nest member, it sets
+				 * a NoClassDefFoundError
+				 */
 				goto _done;
-			} else if (nestMember->nestHost != nestHost) {
+			} else if (NULL == nestMember->nestHost) {
+				/* If loadAndCheckNestHost fails to set the nest member's nest
+				 * host (due to loading or verification error), then it sets an
+				 * IncompatibleClassChangeError
+				 */
+				loadAndCheckNestHost(currentThread, nestMember, TRUE);
```

Comment:
use `loadAndCheckNestHost` consistently.

Reply:
What is the inconsistency here? I use `loadAndCheckNestHost` on each nest member class in order to ensure that their nest host is loaded, prior to checking it against `this`'s nest host. The returned value does not matter in this case

--

This current setup of `loadAndCheckNestHost` is intentional. It allows the same `loadAndCheckNestHost` function to be used for `getNestHost`, `getNestMembers`, and `isNestMateOf` where `getNestMembers` is the only java.lang.Class nestmates method that can set an ICCE (see: http://cr.openjdk.java.net/~dholmes/8010319/specs/java.lang/java/lang/Class.html)

In particular, it allows for the case where:
- Class NestHostA claims nest member list 'NestMemberC'
- Class NestHostB claims nest member list 'NestMemberC'
- Class NestMemberC claims nest host class 'NestHostB'

Then we have the following possible situations
- Calling `NestHostA.isNestMateOf(NestMemberC)` should return false without throwing.
- Calling `NestMemberC .getNestHost()` should return `NestHostB` without throwing.
- Calling `NestHostA.getNestMembers()` should throw ICCE
regardless of whether `isNestMateOf` or `getNestHost` have been called.

Thus, the setup for `loadAndCheckNestHost` will return a valid nest host if possible and return the passed class if no valid nest host is set. The return value is used by `getNestHost` and `isNestmateOf` but is ignored by `getNestMembers`. The set nestmate field is ignored by `getNestHost` and `isNestmateOf` and used by `getNestMembers`.

Reply:
> What is the inconsistency here?
inconsistency is related to the usage of `loadAndCheckNestHost`. at some places, the `return value` is used. at others, `clazz->nestHost` is used. there needs to be a comment in order to clarify why the `return value` is ignored.